### PR TITLE
doc: west 0.12 docs

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Update the documentation for this release.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

### CODE REVIEW

**Comment by mbolivar-nordic** [comment]
DNM because it's still an alpha 3 release at this point; the final release hasn't been cut yet.

**Review by carlescufi — APPROVED** [review]
Reads great!

**Comment by mbolivar-nordic** [comment]
Release is cut. Removed DNM.

File: doc/guides/west/manifest.rst [hunk 1]
```diff
@@ -1054,6 +1058,44 @@ Here, ``west update`` will recursively initialize and update just the
...
+.. code-block:: yaml
+
+   manifest:
+     projects:
+       - name: foo
+       - name: bar
+         userdata: a-string
+       - name: baz
+         userdata:
+           key: value
```
Comment (marc-hb) [inline doc/guides/west/manifest.rst:1085]: Nit:
```suggestion
           some_key: some_value
```
(to save all of us YAML noobs some unnecessary doubt)

File: doc/guides/west/manifest.rst [hunk 2]
```diff
@@ -1054,6 +1058,44 @@ Here, ``west update`` will recursively initialize and update just the
...
+
+.. code-block:: python
+
+   manifest = west.manifest.Manifest.from_file()
+
+   foo, bar, baz = manifest.get_projects(['foo', 'bar', 'baz'])
+
+   foo.userdata # None
+   bar.userdata # 'a-string'
+   baz.userdata # {'key': 'value'}
```
Comment (marc-hb) [inline doc/guides/west/manifest.rst:1097]: See above.

**Review by marc-hb — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**